Source/JavaScriptCore/b3/air/AirAllocateRegistersByGreedy.cpp

```diff
@@ -1752,11 +1808,155 @@ class GreedyAllocator {
+    {
+        auto& useDefs = m_useDefLists[tmp].useDefs();
+
+        Point start = pointAtOffset(interval.begin(), PointOffsets::Pre);
+        Point end = interval.end();
+
+        ASSERT(useDefs.isEmpty() || !cursor || useDefs[cursor - 1] < start);
+
+        size_t i = cursor;
+        for (; i < useDefs.size(); i++) {
+            if (start <= useDefs[i])
+                break;
+        }
+        if (i == useDefs.size() || end <= useDefs[i]) {
+            cursor = i;
+            return { }; // No more use/defs in `interval`
+        }
+
+        BasicBlock* block = findBlockContainingPoint(useDefs[i]);
+        Point positionOfHead = this->positionOfHead(block);
+        Point positionOfTail = this->positionOfTail(block);
+        Point last = std::min(positionOfTail, end - 1);
+
+        do {
+            func(useDefs[i], block->at(instIndex(positionOfHead, useDefs[i])));
+            i++;
+        } while (i < useDefs.size() && useDefs[i] <= last);
+        cursor = i - 1; // i-1 since the next interval may include this final instruction
+        return { last + 1, end }; // Return the unprocessed portion of interval
+    }
+
+    template<Bank bank>
+    bool trySplitIntraBlock(Tmp tmp, TmpData& tmpData)
+    {
+        if (tmpData.splitMetadataIndex)
+            return false;
+
+        unsigned tmpIndex = AbsoluteTmpMapper<bank>::absoluteIndex(tmp);
+        if (bank == GP && m_useCounts.isConstDef<bank>(tmpIndex))
+            return false; // Constant will be rematerialized instead
+
+        BasicBlock* startBlock = findBlockContainingPoint(tmpData.liveRange.intervals().first().begin());
+        Point last = tmpData.liveRange.intervals().last().end() - 1;
+        if (last <= positionOfTail(startBlock))
+            return false; // Tmp's live range is already block local
+
+        ensureUseDefLists();
+
+        SplitMetadata* metadata = nullptr;
+        Vector<Tmp*, 8> tmpPtrs;
+        size_t cursor = 0;
+
+        size_t numIntervals = tmpData.liveRange.intervals().size();
+        for (size_t i = 0; i < numIntervals; i++) {
+            // Note: this loop calls newTmp() which invalidates the tmpData reference.
+            Interval interval = *(m_map[tmp].liveRange.intervals().begin() + i);
+            Interval remaining = interval;
+
+            do {
+                Point lastDefPoint = 0;
+                Interval cluster = { };
+                tmpPtrs.shrink(0);
+
+                remaining = forEachUseDefWithin(tmp, remaining, cursor, [&](Point point, Inst& inst) {
+                    inst.forEachTmp([&](Tmp& t, Arg::Role role, Bank, Width) {
+                        if (t == tmp && !Arg::isColdUse(role)) {
+                            Point early = point + PointOffsets::Early;
+                            tmpPtrs.append(&t);
+                            if (Arg::isAnyDef(role))
+                                lastDefPoint = early; // Remember where the fixup store to spill is needed
```

Comment:
What if you encounter multiple of these in the same forEachDefUseWithin? It seems like you'd only emit a fixup store at the last def.